pkg/connection/store/external/store.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright 2023 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package external implements a gRPC client for external secret store plugins.
+package external
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"path/filepath"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	v1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	ess "github.com/crossplane/crossplane-runtime/apis/proto/v1alpha1"
+	"github.com/crossplane/crossplane-runtime/pkg/connection/store"
+	"github.com/crossplane/crossplane-runtime/pkg/errors"
+)
+
+// Error strings.
+const (
+	errGet    = "cannot get secret"
+	errApply  = "cannot apply secret"
+	errDelete = "cannot delete secret"
+)
+
+// SecretStore is an External Secret Store.
+type SecretStore struct {
+	client     ess.ExternalSecretStoreServiceClient
+	kubeClient client.Client
+	config     *v1.Config
+}
+
+// NewSecretStore returns a new External SecretStore.
+func NewSecretStore(ctx context.Context, kube client.Client, tlsConfig *tls.Config, cfg v1.SecretStoreConfig) (*SecretStore, error) {
+	if cfg.Plugin.Endpoint == nil {
+		return nil, errors.New("endpoint is not provided")
+	}
+
+	creds := credentials.NewTLS(tlsConfig)
+	conn, err := grpc.Dial(*(cfg.Plugin.Endpoint), grpc.WithTransportCredentials(creds))
+	if err != nil {
+		return nil, errors.Wrap(err, fmt.Sprintf("cannot dial to the endpoint: %s", *cfg.Plugin.Endpoint))
+	}
+
+	cl := ess.NewExternalSecretStoreServiceClient(conn)
+
+	return &SecretStore{
+		kubeClient: kube,
+		client:     cl,
+		config:     cfg.Plugin.ConfigRef,
+	}, nil
+}
+
+// ReadKeyValues reads and returns key value pairs for a given Secret.
+func (ss *SecretStore) ReadKeyValues(ctx context.Context, n store.ScopedName, s *store.Secret) error {
+	sec := new(ess.Secret)
```

Comment:
nit:

```suggestion
	sec := &ess.Secret{}
```

I don't think there is something related in the styling docs/guides we are following, but I would like to mention that I am not familiar `new` keyword in XP codebase. You may consider using the suggested way just to be consistent.

Reply:
I _think_ `new` is actually preferred by Go (citation needed) but we tend to use `&Foo{}` syntax instead as you point out. We're committed now so good to stay consistent. :)